src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -311,6 +343,10 @@ NamedIntrinsic HWIntrinsicInfo::lookupId(Compiler*         comp,
         {
             isa = InstructionSet_AVX2;
         }
+        else if (strcmp(className, "Vector512") == 0)
+        {
+            isa = InstructionSet_AVX512F;
```

Comment:
This isn't quite right and may need special handling.

We really need to be checking for both `AVX512BW` and `AVX512DQ` since you need both of them in addition to `F` to fully accelerate `Vector512`. Even if we're unlikely to encounter hardware without support, a user may still set `DOTNET_EnableAVX512BW=0` and we'd want to report `false` in that scenario.

I didn't see anything in `CD` that should be considered a blocker for `Vector512` acceleration, so we likely don't need to also check for that.

Reply:
Tricky. How about the following

```
        else if (strcmp(className, "Vector512") == 0)
        {
            isBaselineVector512IsaUsedAndSupported = comp->IsBaselineVector512IsaSupported();
        }
```


And then modify the check to be


```
            if (comp->compExactlyDependsOn(isa) || isBaselineVector512IsaUsedAndSupported))
            {
                return NI_IsSupported_True;
            }
```

Reply:
Updated with this. Will modify later if needed.